### PR TITLE
CXX-736 - Split out tasks in evergreen

### PR DESCRIPTION
**Pull request by tessavitabile**
https://evergreen.mongodb.com/version/56707e533ff1221cc50003fb_0

### CODE REVIEW

File: .mci.yml [hunk 1]
```diff
@@ -47,42 +47,77 @@ functions:
...
+            keys:
+              - local_key: "aws_key"
+                remote_key: "project_aws_key"
+              - local_key: "aws_secret"
+                remote_key: "project_aws_secret"
+        - command: shell.exec
+          params:
+            script: |
+              rm -rf "mongo-cxx-driver"
+              rm -fr "mongo-c-driver"
```
Comment (markbenvenuto) [inline .mci.yml]: use `rf` instead of `fr`

File: .mci.yml [hunk 2]
```diff
@@ -100,7 +135,10 @@ buildvariants:
       run_on:
           - ubuntu1404-test
       tasks:
-          - name: compile_and_test
+          - name: compile
```
Comment (samantharitter) [inline .mci.yml]: Can you define a variable with this list of tests and use it for both buildvariants?  Would look something like this:

```
tasks: &all_tasks
- compile
- run_examples
- etc

- name: ubuntu-test
  tasks: *all_tasks

- name: ubuntu-debug
  tasks: *all_tasks
```

**Comment by tessavitabile** [comment]
The non-compile tasks have been failing arbitrarily with the message:
Command failed: chdir /data/mci/mongo-cxx-driver/build: no such file or directory
Does anyone have an idea of why this might be happening, and why it only happens sometimes?

**Comment by acmorrow** [comment]
@tessavitabile did the non-compile tasks execute on the same instance where the compile ran?

**Comment by tessavitabile** [comment]
How can I tell? An example is here:
https://evergreen.mongodb.com/build/mongodb_cpp_driver_dev_ubuntu1404_debug_05459816cae6258bad01d248574629795b7d398e_15_12_16_18_05_11

**Comment by acmorrow** [comment]
They definitely ran on different hosts. Check the host for each task that it ran.

**Comment by tessavitabile** [comment]
@jbreams thought it might not be worthwhile to split up the tasks, since the tasks each run quickly, and if we split up the different tasks, then we need to install the C driver and start a mongod each time. Thoughts? He also suggested that if we don't split up the tasks, then we might want to create a way to see which subtask failed, even though they're all part of the same task, which has been done for server tests.

**Comment by acmorrow** [comment]
OK, I'm convinced. Lets close this, I don't think we want to do this, at least not right now.

**Comment by samantharitter** [comment]
Agreed!

**Pull request closed without merging**